### PR TITLE
feat(quickcheck): context parameter for counterexample descriptions

### PR DESCRIPTION
**Pull request by bobzhang**
Remaining item from the #3955 feedback (https://github.com/moonbitlang/core/pull/3955#issuecomment-5189971358, point 3). Until now, attaching human-readable context to a failure (e.g. the rendered TOML document rather than its AST, as moonbit-community/toml-parser#122 needs) required raising `fail(message)` from the property — which lands in `Raised` and conflates "property errored" with "property failed, with context".

- `report()`/`check()` accept `context? : (A) -> String`, evaluated on the **shrunk** counterexample (the covering test shows it rendering the shrink boundary value, not the originally failing input).
- `Falsified`/`Raised` carry `context~ : String?`; the text failure report gains a `context:` line, and the `Debug` rendering includes `context=` only when present — **existing snapshots are untouched** (the full suite passed with no `-u` before the new tests were added).

Test plan: falsified-with-context and raised-with-context report snapshots, a white-box `failure_report` test, and a README example; `moon test quickcheck` 71/71; full core suite green; `moon check --deny-warn`, `moon fmt`, `moon info` clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by CAIMEOX — APPROVED** [review]

File: quickcheck/driver.mbt [hunk 1]
```diff
@@ -113,6 +116,10 @@ pub impl[A : @debug.Debug] @debug.Debug for QuickCheckReport[A] with fn to_repr(
         "Falsified",
         [
           (Some("counterexample"), Repr(counterexample)),
+          ..match context {
```
Comment (bobzhang) [inline quickcheck/driver.mbt]: .. if context is Some(text) { ...}

File: quickcheck/driver.mbt [hunk 2]
```diff
@@ -135,6 +143,10 @@ pub impl[A : @debug.Debug] @debug.Debug for QuickCheckReport[A] with fn to_repr(
         "Raised",
         [
           (Some("counterexample"), Repr(counterexample)),
+          ..match context {
+            Some(text) => [(Some("context"), @debug.Repr(text))]
```
Comment (bobzhang) [inline quickcheck/driver.mbt]: same
